excellent/errors.go

```diff
@@ -0,0 +1,46 @@
+package excellent
```

Comment:
figured while I was reorganizing everything I'd split evaluator.go into evaluator.go, scanner.go and errors.go